http/bucket_service.go

```diff
@@ -353,27 +343,42 @@ type postBucketRequest struct {
 	RetentionRules      []retentionRule `json:"retentionRules"`
 }
 
-func (b postBucketRequest) Validate() error {
+func (b *postBucketRequest) OK() error {
 	if !b.OrgID.Valid() {
 		return &influxdb.Error{
 			Code: influxdb.EInvalid,
-			Msg:  "bucket requires an organization",
+			Msg:  "organization id must be provided",
 		}
+	}
 
+	// Only support a single retention period for the moment
+	var errs []string
+	if len(b.RetentionRules) > 0 {
+		if _, err := b.RetentionRules[0].RetentionPeriod(); err != nil {
+			errs = append(errs, err.Error())
+		}
 	}
+
+	// names starting with an underscore are reserved for system buckets
+	if err := validBucketName(b.toInfluxDB()); err != nil {
+		errs = append(errs, err.Error())
+	}
+
+	if len(errs) > 0 {
+		return &influxdb.Error{
+			Code: influxdb.EUnprocessableEntity,
+			Msg:  strings.Join(errs, "; "),
```

Comment:
How common is this pattern for http error responses? It seems like it would be less computationally intensive (especially if we have validation checks that hit other services or a db) to just do an early return with the first error, and I could see this being difficult to read if there are a lot of errors.

Reply:
in go, unfortunately, its all to uncommon to return early. From a user perspective, i want everything that failed validation. We don't have a mechanism for this atm, which is a pain point for me. Takes multiple requests potentially to figure out what all is wrong with a bucket req in this case. Computationally... its a nothing burger 🤷‍♂. If we are relying on multiple services for **validation**, something has gone horribly wrong with our system architecture 😳 . In that case its even better not to return early imo, otherwise you put stress on the entire system for each additional unneeded request 🤷‍♂. 

This `: ` delimiter is definitely pretty lame. I want something better for sure. But the single error message isn't a whole lot better.  i would love to see something like:

```json
{
    code: "unprocessable entity"
    errors: [
       {
          "field": "name",
          "reason": "name must not be prefixed with \"_\""
       },
       {
          "field": "retentionRules",
          "reason": "expiration seconds must be greater than or equal to one second"
       }
    ]
}
```

Reply:
I opted not to introduce that as it would be a larger chagne, and would require changes on the FE side. Although, it would make user feedback in the UI programmatic at that point if we did. More than just a toast message would do.  @hoorayimhelping you dig?

Reply:
I'd much rather have errors returned as an object than as glommed-together messages, but that'd be a big change. We should get buy-in from @desa before making this change.

Reply:
sounds good, @desa what say you?

Reply:
I'll switch it back to returning early for now. When we get buy in from everyone we can improve this with a collection of failed validations, maybe something like the above 🤷‍♂

Reply:
Would be really awesome to have

```
{
    "code": "unprocessable entity",
    "errors": [
       {
          "field": "name",
          "reason": "name must not be prefixed with \"_\""
       },
       {
          "field": "retentionRules",
          "reason": "expiration seconds must be greater than or equal to one second"
       }
    ]
}
```

I'm onboard with returning all issues with the request.

Reply:
I'll write up a feature req for adding enhanced error return for this situation

Reply:
new issue: https://github.com/influxdata/influxdb/issues/16727

Reply:
I can approve this once the error handling changes are moved out of the PR. We can tackle those changes in ^